File: back-end/norlan-client-portal-master/src/main/java/it/norlan/clientportal/model/Docente.java
package it.norlan.clientportal.model;

import jakarta.persistence.*;
import lombok.*;

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Entity
@Table(name = "docente")
@PrimaryKeyJoinColumn(name = "id_utente")
public class Docente extends Utente {

    @Column(nullable=false)
    private String nome;

    @Column(nullable = false)
    private String cognome;

    @Column(name = "specializzazione_tecnica")
    private String specializzazioneTecnica;
}

File: front-end/Norlan-Svelte-TypeScript-master/src/lib/models/Docente.ts
export interface DocenteData {
	idUtente: number;
	nome: string;
	cognome: string;
	titolo: string;
	email: string;
	specializzazioneTecnica: string;
	bio: string;
}

export class Docente {
	idUtente: number;
	nome: string;
	cognome: string;
	titolo: string;
	email: string;
	specializzazioneTecnica: string;
	bio: string;

	constructor(data: DocenteData) {
		this.idUtente = data.idUtente;
		this.nome = data.nome;
		this.cognome = data.cognome;
		this.titolo = data.titolo;
		this.email = data.email;
		this.specializzazioneTecnica = data.specializzazioneTecnica;
		this.bio = data.bio;
	}
}